.changeset/headersfunctionargs.md

```diff
@@ -0,0 +1,22 @@
+---
+"remix": patch
+"@remix-run/cloudflare": patch
+"@remix-run/deno": patch
+"@remix-run/dev": patch
+"@remix-run/eslint-config": patch
+"@remix-run/node": patch
+"@remix-run/serve": patch
+"@remix-run/server-runtime": patch
+---
+
+add `HeadersFunctionArgs` type to be consistent with loaders/actions/meta and allows for using `function`
+
+```tsx
+import type { HeadersFunctionArgs } from '@remix-run/*';
+
+export function headers({ loaderHeaders }: HeadersFunctionArgs) {
```

Comment:
Maybe use `HeadersArgs` here, as we're using `ActionArgs`/`LoaderArgs` in other places as well?

```suggestion
import type { HeadersArgs } from '@remix-run/node'; // or cloudflare/deno

export function headers({ loaderHeaders }: HeadersArgs) {
```

Reply:
He's added HeadersArgs as an alias, while keeping HeadersFunctionArgs for consistency with DataFunctionArgs

Reply:
Docs/examples/templates are using `ActionArgs`/`LoaderArgs` though, hence why I think we should use that in the changelog as well